cli/azd/pkg/infra/azure_resource_manager.go

```diff
@@ -30,6 +30,7 @@ type ResourceManager interface {
 		resourceType AzureResourceType,
 	) (string, error)
 	GetWebAppResourceTypeDisplayName(ctx context.Context, subscriptionId string, resourceId string) (string, error)
+	GetCognitiveServiceResourceTypeDisplayName(ctx context.Context, subscriptionId string, resourceId string) (string, error)
```

Comment:
In my opinion  we should not be adding these new funcs to the interface since they are internal implementation details on how to get the display name.  

My recommendation would be to remove both `GetCognitiveServiceResourceTypeDisplayName` and `GetWebAppResourceTypeDisplayName` from the interface and rename to be private functions.